MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.swift

```diff
@@ -41,25 +41,48 @@
 import UIKit
 
 protocol TransactionsTableViewDelegate {
-    func onTransactionSelect(_: Transaction)
+    func onTransactionSelect(_: Any)
     func onScrollDirectionChange(_: ScrollDirection)
 }
 
 class TransactionsTableViewController: UITableViewController {
     let CELL_IDENTIFIER = "TransactionTableTableViewCell"
-    let transactions = dummyTransactions
     var actionDelegate: TransactionsTableViewDelegate?
     var refreshTransactionControl = UIRefreshControl()
     private var lastContentOffset: CGFloat = 0
     private var lastScrollDirection: ScrollDirection = .up
 
+    var groupedCompletedTransactions: [[CompletedTransaction]] = []
+    var pendingInboundTransactions: [PendingInboundTransaction] = []
+    var pendingOutboundTransactions: [PendingOutboundTransaction] = []
+
+    var showPendingGroup: Bool {
+        return pendingInboundTransactions.count > 0 || pendingOutboundTransactions.count > 0
+    }
+
+    var showEmptyState: Bool = false {
+        willSet {
+            //Stop it from getting re added each time unnecessarily
+            if newValue && !showEmptyState {
+                setEmptyView()
+            } else if !newValue {
+                removeEmptyView()
+            }
+        }
+    }
+
+    var tempRefreshTimer = Timer()
```

Comment:
Lowest possible visibility (access-control) is good practice. I think you can use private here, and for all similar variables/values/etc.